Packages/com.unity.multiplayer.samples.coop/Tests/Runtime/HostAndDisconnectTest.cs

```diff
@@ -0,0 +1,265 @@
+using System;
+using System.Collections;
+using NUnit.Framework;
+using Unity.Multiplayer.Samples.BossRoom.Client;
+using Unity.Multiplayer.Samples.BossRoom.Shared;
+using Unity.Multiplayer.Samples.BossRoom.Shared.Net.UnityServices.Lobbies;
+using Unity.Multiplayer.Samples.BossRoom.Visual;
+using Unity.Multiplayer.Samples.Utilities;
+using Unity.Netcode;
+using Unity.Services.Authentication;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.TestTools;
+
+namespace Unity.Multiplayer.Samples.BossRoom.Tests.Runtime
+{
+    public class HostAndDisconnectTest
+    {
+        const string k_BootstrapSceneName = "Startup";
+
+        const string k_MainMenuSceneName = "MainMenu";
+
+        const string k_CharSelectSceneName = "CharSelect";
+
+        const string k_BossRoomSceneName = "BossRoom";
+
+        static int[] s_PlayerIndices = new int[] { 0, 1, 2, 3, 4, 5, 6, 7 };
+
+        NetworkManager m_NetworkManager;
+
+        const float k_ServiceQueryTimeout = 10f;
+
+        /// <summary>
+        /// Smoke test to validating hosting inside Boss Room. The test will load the project's bootstrap scene,
+        /// Startup, and commence the game flow as a host, pick and confirm a parametrized character, and jump into the
+        /// BossRoom scene, where the test will disconnect the host.
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator Lobby_HostAndDisconnect_Valid([ValueSource(nameof(s_PlayerIndices))] int playerIndex)
+        {
+            yield return GoToMainMenuScene();
+
+            // now inside MainMenu scene
+
+            yield return null;
+
+            var clientMainMenuState = GameObject.FindObjectOfType<ClientMainMenuState>();
+            Assert.That(clientMainMenuState != null, $"{nameof(clientMainMenuState)} component not found!");
+
+            var scope = clientMainMenuState.DIScope;
+
+            var lobbyUIMediator = scope.Resolve<LobbyUIMediator>();
+            Assert.That(lobbyUIMediator != null, $"{nameof(LobbyUIMediator)} component not found!");
+
+            var lobbyCreationUI = lobbyUIMediator.LobbyCreationUI;
+            Assert.That(lobbyCreationUI != null, $"{nameof(LobbyCreationUI)} component not found!");
+
+            // validate unity services login successful
+
+            // wait until authenticated
+            var timer = k_ServiceQueryTimeout;
+            while (timer > 0f && !AuthenticationService.Instance.IsAuthorized)
+            {
+                timer -= Time.deltaTime;
+                yield return null;
+            }
+
+            clientMainMenuState.OnStartClicked();
+
+            yield return new WaitForEndOfFrame();
+
+            lobbyUIMediator.ToggleCreateLobbyUI();
```

Comment:
I'm noticing you're doing a lot of these manual steps. Shouldn't clicking the Start button do this for you?

Reply:
If you're doing these manual steps, then why not just trigger hosting using GameNetPortal and go from there?

Reply:
as discussed, some comments on what these do would be great :)